src/arrayMethodJoin.js

```diff
@@ -4,8 +4,33 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinResultString = '';
+
+    if (this.length < 0) {
+      return joinResultString;
+    }
+
+    let workSeparator = separator;
+
+    if (workSeparator === null) {
+      workSeparator = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      let element = this[i];
+
+      if (element === null || element === undefined) {
+        element = '';
+      }
+
+      if (separator !== undefined && this.length > 1 && i !== this.length - 1) {
```

Comment:
```suggestion
      if (i !== this.length - 1) {
```
the separator will never be undefined because you gave the default value for it.
The second check is aslo redundant